thunder/dynamo/report.py

```diff
@@ -996,26 +1004,103 @@ def foo(x):
+            print(f"Failed to run the function with exception: {e}")
+            return
+        print("The input callable can be successfully executed in eager mode.")
+    else:
+        print("The input callable can be successfully executed by torch.compile.")
+
+
+def get_thunder_fxgraph_reports(
+    fn: Callable, *args, thunder_compile_kwargs: dict = None, check_runnablility=True, **kwargs
+) -> list[ThunderFXGraphReport]:
+    """
+    Generates a list of :class:`ThunderFXGraphReport` objects for a given callable.
+
+    This function performs the following steps:
+    1. Checks if the callable can be executed with `torch.compile` or eager execution when `check_runnablility` is `True`.
+    2. Generates the dynamo segmented FX graphs and further analyzes the Thunder-split subgraphs of the FX graph.
+    """
+    if check_runnablility:
+        check_torch_runnablility(fn, *args, **kwargs)
+
     reports = fx_report(fn, *args, **kwargs)
     if thunder_compile_kwargs is None:
         thunder_compile_kwargs = {}
-    split_reports = []
-    for fx_graph_report in reports.fx_graph_reports:
-        thunder_fx_graph_report = analyze_thunder_splits(fx_graph_report, **thunder_compile_kwargs)
-        print(f"\n{thunder_fx_graph_report.graph_name}: {thunder_fx_graph_report.split_reason}\n")
-        for split_report in thunder_fx_graph_report.subgraph_reports:
-            split_reports.append(split_report)
-    return split_reports
+    thunder_fxgraph_reports = []
+    for fxgraph_report in reports.fx_graph_reports:
+        thunder_fxgraph_report = analyze_thunder_splits(fxgraph_report, **thunder_compile_kwargs)
+        thunder_fxgraph_reports.append(thunder_fxgraph_report)
+    return thunder_fxgraph_reports
 
 
-def get_nvfusion_reports(split_reports: list[ThunderSplitGraphReport]):
-    nvfusion_reports = []
+def make_nvfusion_reports(split_reports: list[ThunderSplitGraphReport]):
+    """
+    Creates nvFusion reports for each Thunder-split report.
+    """
     for split_report in split_reports:
         split_report.create_fusion_reports()
-        for nvfusion_report in split_report.fusion_reports:
-            nvfusion_reports.append(nvfusion_report)
-    return nvfusion_reports
+
+
+# NOTE: Here we use TorchInductorSpecification (TorchInductor without TorchDynamo) to compare with Thunder. Reason: https://github.com/Lightning-AI/lightning-thunder/issues/1521
+def thunderfx_benchmark_report_from_splits(
+    thunder_fxgraph_reports: list[ThunderFXGraphReport],
+    folder_path: str | PathLike,
+    thunder_compile_kwargs: dict = None,
+    compare_fusion: bool = False,
+    rtol=0.5,
+    atol=0.0,
+):
+    """
+    A utility function that analyzes the runnability and performance benchmarks of each Thunder-split FX graph and nvFusion regions(optional). It prints out the performance metrics and saves the benchmark script in `folder_path` if the difference exceeds the tolerance (`rtol`, `atol` in seconds).
```

Comment:
This might create multiple files, right?

Would you elaborate on the directory structure and name of the files this creates? 

Not for this PR, but in the future should we think about the filenames including a short description of why the file was generated? Something like:

```
run09_graph13_speed_bench.py
```

?

Reply:
Definitely we can't include the full reason -- let's not worry about it for now